src/core/src/test/java/org/locationtech/geogig/test/integration/MergeOpTest.java

```diff
@@ -576,6 +572,69 @@ public void testMergeFastForwardSecondCase() throws Exception {
     }
 
     @Test
+    public void testFastForwardOnly() throws Exception {
+        // Create the following revision graph
+        // o
+        // |
+        // o - Point 1 added
+        // |\
+        // | o - TestBranch - Point 1 modified
+        // |
+        // o - master - HEAD - Point 1 modified
+        insertAndAdd(points1);
+        geogig.command(CommitOp.class).call();
+        geogig.command(BranchCreateOp.class).setName("TestBranch").call();
+        Feature points1Modified = feature(pointsType, idP1, "StringProp1_2", new Integer(1000),
+                "POINT(1 1)");
+        insertAndAdd(points1Modified);
+        geogig.command(CommitOp.class).call();
+        geogig.command(CheckoutOp.class).setSource("TestBranch").call();
+        Feature points1ModifiedB = feature(pointsType, idP1, "StringProp1_3", new Integer(2000),
+                "POINT(1 1)");
+        insertAndAdd(points1ModifiedB);
+        geogig.command(CommitOp.class).call();
+        geogig.command(CheckoutOp.class).setSource("master").call();
+        Ref branch = geogig.command(RefParse.class).setName("TestBranch").call().get();
+        MergeOp mergeOp = geogig.command(MergeOp.class).addCommit(Suppliers.ofInstance(branch.getObjectId()));
+        mergeOp.setFastForwardOnly(true);
+        try {
```

Comment:
Check there's a JUnit ExpectedException rule at the top of the file. Please use it instead of try/fail(). Something like:
exception.expect(IllegalStateException.class);
exception.expectMessage(<known substring>);
mergeOp.call();